huaweicloud/services/tms/resource_huaweicloud_tms_resource_tags.go

```diff
@@ -0,0 +1,268 @@
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceResourceTags() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceResourceTagsCreate,
+		ReadContext:   resourceResourceTagsRead,
+		UpdateContext: resourceResourceTagsUpdate,
+		DeleteContext: resourceResourceTagsDelete,
+
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The project ID of the resources.",
+			},
+			"resources": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"resource_type": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The resource type.",
+						},
+						"resource_id": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The resource ID.",
+						},
+					},
+				},
+				Description: "The managed resource configuration.",
+			},
+			"tags": {
+				Type:     schema.TypeMap,
+				Required: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Description: "The resource tags for batch management.",
+			},
+		},
+	}
+}
+
+func buildResourcesInfo(resources []interface{}) []tags.Resource {
+	if len(resources) < 1 {
+		return nil
+	}
+
+	result := make([]tags.Resource, len(resources))
+	for i, val := range resources {
+		resource := val.(map[string]interface{})
+		result[i] = tags.Resource{
+			ResourceType: resource["resource_type"].(string),
+			ResourceId:   resource["resource_id"].(string),
+		}
+	}
+	return result
+}
+
+func expandResourceTags(tagsInput map[string]interface{}) []tags.ResourceTag {
+	result := make([]tags.ResourceTag, 0, len(tagsInput))
+
+	for key, value := range tagsInput {
+		result = append(result, tags.ResourceTag{
+			Key:   key,
+			Value: utils.String(value.(string)),
+		})
+	}
+	return result
+}
+
+func resourceResourceTagsCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := meta.(*config.Config).TmsV1Client()
+	if err != nil {
+		return diag.Errorf("error creating TMS v1 client: %s", err)
+	}
+
+	opts := tags.BatchOpts{
+		ProjectId: d.Get("project_id").(string),
+		Resources: buildResourcesInfo(d.Get("resources").([]interface{})),
+		Tags:      expandResourceTags(d.Get("tags").(map[string]interface{})),
+	}
+	_, err = tags.Create(client, opts)
+	if err != nil {
+		return diag.Errorf("error creating resource tags: %s", err)
+	}
+
+	randUUID, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate resource ID of the TMS tags management: %s", err)
+	}
+	d.SetId(randUUID)
+
+	return resourceResourceTagsRead(ctx, d, meta)
+}
+
+func FlattenTagsToMap(tagsResp []tags.ResourceTag) map[string]interface{} {
+	result := make(map[string]interface{})
+	for _, val := range tagsResp {
+		result[val.Key] = val.Value
```

Comment:
can we return map[string]string?

```suggestion
		result[val.Key] = *val.Value
```

Reply:
done.